### PR TITLE
LSP: code reorganize to support multiple projects simultaneously

### PR DESCRIPTION
**Pull request by JoshuaBatty**
The main change is instead of having a single `session: Arc<Session>`, we now have a `sessions: DashMap<PathBuf, Arc<Session>>,`. The `PathBuf` is the manifest directory of the sway project and the `Session` contains all the tokens, runnables etc.. for that project.

Although we are blocked currently to support multiple simultaneous projects, this PR fixes a current bug that now allows the users to jump between different projects in a workspace so long as they only have one window pane open. 

related: #3069 and #2063

### CODE REVIEW

**Review by otrho — APPROVED** [review]
Fair enough.

File: sway-lsp/src/server.rs [hunk 1]
```diff
@@ -124,15 +115,55 @@ fn capabilities() -> ServerCapabilities {
...
+                    session.shutdown();
+                });
+
+                // If no session can be found, then we need to call init and inserst a new session into the map
+                self.init(uri)?;
+                self.sessions
+                    .get(&manifest_dir) //
+                    .map(|item| item.value().clone())
+                    .expect("no session found even though it was just inserted into the map")
+            }
```
Comment (mitchmindtree) [inline sway-lsp/src/server.rs:151]: It looks like the previous session is not removed from `sessions`, which means if a user switches back the previous session will no longer be active? I guess this is fine for now as the real solution is addressing https://github.com/FuelLabs/sway/issues/2063 so you can actually have multiple simultaneous sessions.

File: sway-lsp/src/server.rs [hunk 2]
```diff
@@ -155,12 +186,13 @@ impl Backend {
 #[tower_lsp::async_trait]
 impl LanguageServer for Backend {
     async fn initialize(&self, params: InitializeParams) -> jsonrpc::Result<InitializeResult> {
+        eprintln!("InitializeParams = {:#?}", params);
```
Comment (mitchmindtree) [inline sway-lsp/src/server.rs:189]: Leftover debug stuff? Or should this use `tracing::trace!`?

**Review by mitchmindtree — APPROVED** [review]

File: sway-lsp/src/server.rs (hunk 1 above)
Comment (JoshuaBatty) [reply sway-lsp/src/server.rs:151]: Good spot, i'm going to just clear the map after shutting down the session in the meantime. Hopefully we can remove this block of code once #2063 lands and we can store multiple sessions.

**Pull request merged**